### DISCUSSION TITLE
AoA for non valid paths

### DISCUSSION THREAD

**Discussion by TheoriginalMMM** [NVlabs/sionna#596] [Q&A] [answered]
Hello,

I noticed that the path properties theta_r and phi_t still have values, even for paths that are marked as invalid (I’ve been using the paths.mask property to verify valid paths, along with the delays paths.tau).

This seems a bit unexpected to me, as I would assume the angles for invalid paths should be assigned special values (e.g., NaN or similar). Could you please explain the logic implemented behind assigning these angle values, even for invalid paths? It’s possible 
that I might have misunderstood something.

To illustrate what i am talking about, given a scene with 10 transmitters and one receiver, i used this code while setting the scene.synthetic array at true.
```py
paths = scene.compute_paths(max_depth=5,los=True,
                            reflection=True,diffraction=True,scattering=True,
                            ris=False,scat_keep_prob=0.001,edge_diffraction=False,
                            check_scene=main_params["check_scene"],scat_random_phases=True)

a_cir,tau_cir = paths.cir(los=True,reflection=True,diffraction=True,scattering=True,
                  ris=False,cluster_ris_paths=False)
                  
print("paths.mask shape :",paths.mask.shape)
print("Paths mask for the link between the user and the first base station :")
print(paths.mask[0,0,0,:14])

print("Delays shape :",paths.tau.shape)
print("Delays paths.cir shape:",tau_cir.shape)
print("Delays paths.cir First link :",tau_cir[0,0,0,:14])

print("Paths delay for the link between the user and the first base station  :",paths.tau[0,0,0,:14])
print("#"*50)
print("AoA Zenith shape :",paths.theta_r.shape)
print("AoA Zenith :",paths.theta_r[0,0,1,:14])
print("AoA Azimuth :",paths.phi_r[0,0,1,:14])
print("#"*50)
print("AoD Zenith :",paths.theta_t[0,0,1,:14])
print("AoD Azimuth :",paths.phi_t[0,0,1,:14])

``` 

I got this output logs:

<details><summary>Details</summary>
<p>

paths.mask shape : (1, 1, 10, 40)
Paths mask for the link between the user and the first base station :
tf.Tensor(
[False False False False False False False False False False False  True
  True  True], shape=(14,), dtype=bool)
Delays shape : (1, 1, 10, 40)
Delays paths.cir shape: (1, 1, 10, 40)
Delays paths.cir First link : tf.Tensor(
[-1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00
 -1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00
 -1.000000e+00  5.545769e-08  0.000000e+00  3.201876e-09], shape=(14,), dtype=float32)
Paths delay for the link between the user and the first base station  : tf.Tensor(
[-1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00
 -1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00
 -1.000000e+00  5.545769e-08  0.000000e+00  3.201876e-09], shape=(14,), dtype=float32)
##################################################
AoA Zenith shape : (1, 1, 10, 40)
AoA Zenith : tf.Tensor(
[1.5175612 1.5175612 1.5175612 1.5175612 1.5175612 1.5175612 1.5175612
 1.5175612 1.5175612 1.5175612 1.5175612 0.1580508 1.5177416 1.1915371], shape=(14,), dtype=float32)
AoA Azimuth : tf.Tensor(
[ 1.9693805  1.9693805  1.9693805  1.9693805  1.9693805  1.9693805
  1.9693805  1.9693805  1.9693805  1.9693805  1.9693805 -1.8089132
  2.2260492  1.9754554], shape=(14,), dtype=float32)
##################################################
AoD Zenith : tf.Tensor(
[1.6240315 1.6240315 1.6240315 1.6240315 1.6240315 1.6240315 1.6240315
 1.6240315 1.6240315 1.6240315 1.6240315 1.586851  1.6238512 1.5902731], shape=(14,), dtype=float32)
AoD Azimuth : tf.Tensor(
[-1.1722122 -1.1722122 -1.1722122 -1.1722122 -1.1722122 -1.1722122
 -1.1722122 -1.1722122 -1.1722122 -1.1722122 -1.1722122 -1.1757162
 -1.1985625 -1.1728071], shape=(14,), dtype=float32)

</p>
</details> 

Thank you in advance for your help!

**Comment by jhoydis** [comment, marked as answer]
Hi @TheoriginalMMM,

You are right. We do not set all attributes of invalid paths to special values as they are masked and carry zero energy.
That means that you should only rely on the mask property to filter out valid paths.

**Reply by TheoriginalMMM** [reply]
Thank you for your quick and clear reply !
